tests/Oryx.BuildImage.Tests/DotNetCore/DotNetCoreSampleAppsTest.cs

```diff
@@ -938,7 +931,7 @@ public void DotNetCore_Muxer_ChoosesAppropriateSDKVersion(string sdkversion)
             var script = new ShellScriptBuilder()
                 .AddCommand($"mkdir -p {appDir} && cd {appDir}")
                 .AddCommand($"dotnet new globaljson --sdk-version {sdkversion}")
-                .SetEnvironmentVariable("PATH", $"{flattenedDotNetInstallDir}:$PATH")
+                .SetEnvironmentVariable("PATH", $"{flattenedDotNetInstallDir}:$PATH", true)
```

Comment:
@waliMSFT do you mind clarifying why this `true` param was added to this `SetEnvironnmentVariable` method and what it means?

Reply:
yah sure. I actually added few lines of comment in the definition of the `SetEnvironnmentVariable` method ([url](https://github.com/microsoft/Oryx/pull/1878/files/3840c7f9d114fa8a6d0b706eca5842985f933c4a#diff-553be68e282a55d647d81b70373bbc10beeb087b9e673a1531a89bda3889ec79:~:text=//%20linux%20does%20not,substitution%20is%20required.)). 

The reason to add the boolean value is we need to use two different kinds of quotations: single and double. `', "`. The reason:
1) when the sas-token is set as ENV variable we need to enquote the token value with a single quote. Otherwise, the Linux system cannot handle the special characters within the token value.
2) Except one special case this single quoted value works fine. In a situation where substitution is used to construct a string (for example: `"{flattenedDotNetInstallDir}:$PATH" where flattenedDotNetInstallDir is a path to a directory`) Linux system doesn't allow the variable substitution inside a single quote. Variable substitution is only allowed inside a double quoted string value.